### PR TITLE
Quality Audit Page Complete

### PR DESCRIPTION
**Pull request by RevatureGentry**
Quality audit page now properly tracks all user actions and persists them properly. Began creation of environment.ts file for more maintainable services. Also implemented Batch Select dropdown menu with search capabilities

### CODE REVIEW

File: src/app/Assess-Batch/Services/category.service.ts [hunk 1]
```diff
@@ -27,10 +27,14 @@ export class CategoryService {
...
-    return this.http.get<[Category]>(`${this.url}/all`, httpOptions);
+    return this.http.get<[Category]>(`${this.url}`, httpOptions);
   }
 
   getCategoryById(id: number): Observable<Category> {
     return this.http.get<Category>(`${this.url}/${id}`, httpOptions);
   }
+
+  getActiveCatgories(): Observable<Category[]> {
+    return this.http.get<Category[]>(environment.api.category.active, httpOptions);
```
Comment (Quinn-Donnelly) [inline src/app/Assess-Batch/Services/category.service.ts]: Question around this. If the plan is to fully remove the URLs from the codebase then line 34 should also have something similar to this. Is that the direction we are going?

Comment (Quinn-Donnelly) [reply src/app/Assess-Batch/Services/category.service.ts]: Or is this just a change until the high level services are done.

**Review by Quinn-Donnelly — APPROVED** [review]
I think this is all good just need to determine if we want to do the environment based url building or if we want to do the gRPC or strings in services with the base changing.

**Review by Quinn-Donnelly — APPROVED** [review]
Build is failing though. We need to make sure this build can run `ng build --prod --aot`

**Review by Quinn-Donnelly — CHANGES_REQUESTED** [review]
See above clicked wrong button

File: src/app/Assess-Batch/Services/category.service.ts (hunk 1 above)
Comment (RevatureGentry) [reply src/app/Assess-Batch/Services/category.service.ts]: Correct, I am normalizing the URL's in the high level services - making the environment file as new methods are added to each respected service

**Comment by RevatureGentry** [comment]
Also just realized that categories this week needs to be on the Quality Audit page. Will fix both issues in next commit

**Review by Quinn-Donnelly — APPROVED** [review]
Excellent improvements.

**Pull request merged**